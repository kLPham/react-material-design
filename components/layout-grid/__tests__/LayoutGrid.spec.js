import React from 'react';
import { shallow, mount } from 'enzyme';
import LayoutGrid from '../index';
import LayoutGridCell from '../layoutGridCell';

describe('<LayoutGrid />', () => {
    it('should render', () => {
        const layoutGrid = shallow(
            <LayoutGrid>
                <LayoutGridCell />
            </LayoutGrid>,
    );
        expect(layoutGrid).toMatchSnapshot();
    });
    it('should render nested', () => {
        const nested = true;
        const layoutGrid = mount(
            <LayoutGrid>
                <LayoutGridCell>
                    <LayoutGrid nested={nested}>
                        <LayoutGridCell><div>Second Level</div></LayoutGridCell>
                        <LayoutGridCell><div>Second Level</div></LayoutGridCell>
                    </LayoutGrid>
                </LayoutGridCell>
                <LayoutGridCell><div>First Level</div></LayoutGridCell>
                <LayoutGridCell><div>First Level</div></LayoutGridCell>
            </LayoutGrid>,
      );
        expect(layoutGrid).toMatchSnapshot();
    });
});
