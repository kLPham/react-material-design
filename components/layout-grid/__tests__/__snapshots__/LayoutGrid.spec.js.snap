// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LayoutGrid /> should render 1`] = `
<div
  className="mdc-layout-grid"
>
  <div
    className="mdc-layout-grid__inner"
  >
    <LayoutGridCell />
  </div>
</div>
`;

exports[`<LayoutGrid /> should render nested 1`] = `
<LayoutGrid>
  <div
    className="mdc-layout-grid"
  >
    <div
      className="mdc-layout-grid__inner"
    >
      <LayoutGridCell>
        <div
          className="mdc-layout-grid__cell"
        >
          <LayoutGrid
            nested={true}
          >
            <div
              className="mdc-layout-grid__inner"
            >
              <LayoutGridCell>
                <div
                  className="mdc-layout-grid__cell"
                >
                  <div>
                    Second Level
                  </div>
                </div>
              </LayoutGridCell>
              <LayoutGridCell>
                <div
                  className="mdc-layout-grid__cell"
                >
                  <div>
                    Second Level
                  </div>
                </div>
              </LayoutGridCell>
            </div>
          </LayoutGrid>
        </div>
      </LayoutGridCell>
      <LayoutGridCell>
        <div
          className="mdc-layout-grid__cell"
        >
          <div>
            First Level
          </div>
        </div>
      </LayoutGridCell>
      <LayoutGridCell>
        <div
          className="mdc-layout-grid__cell"
        >
          <div>
            First Level
          </div>
        </div>
      </LayoutGridCell>
    </div>
  </div>
</LayoutGrid>
`;
